pkg/auth/auth.go

```diff
@@ -64,29 +64,61 @@ func Auth(adminKey, keyString string) (*SignedInUser, error) {
 	// check the cache
 	log.Debug("Checking cache for apiKey")
 	user, cached := cache.Get(keyString)
-	if user != nil {
-		log.Debug("valid key cached")
-		return user, nil
-	}
 	if cached {
+		if user != nil {
+			log.Debug("valid key cached")
+			return user, nil
+		}
+
 		log.Debug("invalid key cached")
 		return nil, ErrInvalidApiKey
 	}
 
-	//validate the API key against grafana.net
+	// validate the API key against grafana.net with a 1s timeout
+	timeout := time.Duration(1 * time.Second)
+	client := http.Client{
```

Comment:
This is not a good idea.  Each "client" is responsible for their own connection pools.  By creating a new client struct for each request, previous connections (assuming KeepAlives are enabled) cant be re-used.